wal_e/retries.py

```diff
@@ -70,6 +71,8 @@ def shim(*args, **kwargs):
                         exc_processor_cxt = exception_processor(
                             exception_info_tuple,
                             exc_processor_cxt=exc_processor_cxt)
+                    except UserCritical:
```

Comment:
This will unfortunately change the semantics of any `UserCritical`. Have you checked all such `raise`s to make sure that they make sense to terminate a retry-loop?